tests/feinterface/step-12.cc

```diff
@@ -300,7 +300,7 @@ namespace Step12
                                ScratchData<dim> &  scratch_data,
                                CopyData &          copy_data) {
       scratch_data.fe_interface_values.reinit(cell, face_no);
-      const FEFaceValuesBase<dim> &fe_face =
+      const FEValuesBase<dim> &fe_face =
```

Comment:
So this change is not backwards-compatible? Hrm.
One could also leave the old type and fail if one calls this function after using the new constructor. What do you think?

Reply:
Unfortunately, yes. The main problem is that `get_fe_face_values()` is called internally to compute jumps, averages, etc. so it must be available also with the new cstr. At the same time, the name `get_fe_face_values()` is now misleading, as in the present PR it's not returning a FEFaceValuesBase anymore, so I agree one should use the old types. 

To be more specific, everything boils down to the fact that `fe_face_values` and `fe_face_values_neighbor` now need to be of type FEValuesBase. 

So far the only solution I see is to add two FEValuesBase* members that will be initialized in case one chooses the new constructor. This would propagate some asserts and if sentences to the code, though.